x-pack/platform/plugins/shared/alerting/server/application/rule/methods/bulk_edit/bulk_edit_rules.ts

```diff
@@ -468,9 +468,15 @@ async function updateRuleAttributesAndParamsInMemory<Params extends RuleParams>(
       attributes: rule.attributes as RawRule,
     });
 
+    const ruleArtifacts = injectReferencesIntoArtifacts(
+      rule.id,
+      rule.attributes.artifacts,
+      rule.references || []
+    );
+
     if (migratedActions.hasLegacyActions) {
       rule.attributes.actions = migratedActions.resultedActions;
-      rule.references = migratedActions.resultedReferences;
+      rule.references = migratedActions.resultedReferences; // TODO check if this will break references that include artifacts
```

Comment:
@cnasikas Now that we add artifacts to the references as well, will this line of code break something? I have the impression it will exclude the artifacts references. Can you confirm if everything will keep working as expected?

Reply:
@cnasikas bumping this to make sure we get this answered.

Reply:
I moved `injectReferencesIntoArtifacts` in this [commit](https://github.com/elastic/kibana/pull/216292/commits/e57875e115e73b249421bbbc2a432cd0f2c67186).